### ISSUE TITLE
System proxy settings(Automatically detect settings) are no longer used in v2.0.

### ISSUE THREAD

**Issue by yuichi-github** [chocolatey/choco#3181]
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my problem.
- [X] I have verified this is not an issue for a specific package.
- [X] I have verified this issue is not security related.

### What You Are Seeing?

System proxy settings where "Automatically detect settings" is enabled are no longer used after upgrading v2.0.
For example, when we run the command `choco upgrade git -y', the error below occurs.

```
Unable to load the service index for source https://chocolatey.org/api/v2/.
Unable to connect to source 'https://chocolatey.org/api/v2/':
object reference not set to an instance of an object
```

https://docs.chocolatey.org/en-us/guides/usage/proxy-settings-for-chocolatey
System proxy settings

### What is Expected?

"Automatically detect settings" work as v1.4 or lower.

### How Did You Get This To Happen?

(above-mentioned)

### System Details

* Operating System: 10.0.19044.0
* Windows PowerShell version: 5.1.19041.2673
* Chocolatey CLI Version: 2.0.0
* Chocolatey Licensed Extension version: ?
* Chocolatey License type: ?
* Terminal/Emulator:

### Installed Packages

```bash
no packages
```

### Output Log

```bash
(above-mentioned)
```

**Comment by gep13** [comment]
@yuichi-github can you please go back and fill in a complete and comprehensive set of steps to reproduce this problem?  The reason that I ask is that there was extensive testing around the use of a Proxy with Chocolatey in this release, and everything that we tested works as expected.  It could be something specific in the proxy that you are using, or the way that it is configured, that is causing the problem.

In order to fix an issue, we need to be able to fully reproduce it.

It would also be very useful to have a complete log provided for the operation that you are performing, not just the snippet that you have provided.

When completing the steps to reproduce, I am not expecting you to provide us with the exact IP address, or host name of your proxy server, but it would be good to know the exact format of the proxy information that you are providing, so that, again, we can attempt to reproduce the problem.

**Comment by KevinRossUWV** [comment]
I have the same issue, downgraded back to 1.4.0 and problems disapears.

Also running `choco outdated` gives the same error

```
Chocolatey v2.0.0
Outdated Packages
 Output is package name | current version | available version | pinned?

Unable to load the service index for source [https://community.chocolatey.org/api/v2/.](https://community.chocolatey.org/api/v2/)
Unable to connect to source 'https://community.chocolatey.org/api/v2/':
 Object reference not set to an instance of an object.
```
Environment:
- Running the chocolatey commands on a Windows Server 2019 machine
- inside a self-hosted Azure Agent job 
- which has a .proxy and .proxybypass configuration. 
- The machine has OS proxy and proxy bypass settings. 
- We have no OS environment variables configured for proxy/proxybypass.
- Agent runs as a service with an account that is local administrator on the machine

**Comment by yuichi-github** [comment]
@gep13 
Thanks for the advice. I updated the message at beginning of thread.

**Comment by ChristianKoessel** [comment]
Same issue for us after upgrade from 1.4 to 2.0:
`Chocolatey v2.0.0
Upgrading the following packages:  
all  
By upgrading, you accept licenses for the packages.  
Unable to load the service index for source https://chocolatey.org/api/v2/.  
Unable to connect to source 'https://chocolatey.org/api/v2/':  
 Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.`

**Comment by pauby** [comment]
@ChristianKoessel is this the exact same issue for you, with the same set of reproducible steps?

@ChristianKoessel @yuichi-github Can you confirm what proxy software you're running and type of proxy and whether it is authenticated or not and type of authencation? If you do not want to provide that in public, can you send me a DM on [Discord](https://ch0.co/community) (I'm also pauby on there).

Thanks.

**Comment by ChristianKoessel** [comment]
Even with explicit proxy settings it didn't work.
I downgraded to 1.4.0 and everything is fine again.
We hava a proxy with NTLM authentication and proxy.pac script.

**Comment by KevinRossUWV** [comment]
> Even with explicit proxy settings it didn't work. I downgraded to 1.4.0 and everything is fine again. We hava a proxy with NTLM authentication and proxy.pac script.

I've just tested `explicit proxy settings` and `explicit proxy settings at runtime`, this did solve it for me. But not what we want :-)

But the difference is we don't have a proxy.pac and no NTLM authentication.

**Comment by pauby** [comment]
@KevinRossUWV @ChristianKoessel Thanks for providing that.

**Comment by mcsmit** [comment]
I have the same issue and we even don't use proxy settings. I also downgraded to 1.4.0 to get things going.

**Comment by pauby** [comment]
@mcsmit if you are not using any proxy settings, can you please raise a separate issue? Please include reproducible steps and log files.

Thanks.

**Comment by corbob** [comment]
So it looks like there's [some configurations](https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-proxy-server-settings) (WPAD and PAC at least) that Windows stores in different places from where we're looking.

**Comment by yuichi-github** [comment]
I updated the top comment with below.

> **Additional Context**
> Proxy: No Authentication / Proxy.pac

**Comment by TheCakeIsNaOH** [comment]
> Since updating from chocolatey itself is not an option due to this bug.

There are two potential options I can think of. 

The first would be to explicitly specify the proxy during upgrade, like `choco upgrade chocolatey --proxy=.... --proxy-user=... --proxy-password=...`

The second would be to download the Chocolatey CLI `.nupkg` file and specify the path to the folder it is in with `--source`, like `choco upgrade chocolatey --source="'D:\download\folder'"` The folder could be a local folder, or it could be on a network drive, either works.

**Comment by gep13** [comment]
@yuichi-github @KevinRossUWV @ChristianKoessel @mcsmit @opusforlife2 and new Chocolatey CLI alpha release has been shipped, which should contain a fix for the issue that has been reported here.  If you are in a position to, it would be great if you could test out this version, and report back with whether things are now working as expected.  Thanks!

https://docs.chocolatey.org/en-us/choco/release-notes#alpha-20230614-june-14-2023

**Comment by mcsmit** [comment]
I would love to test but my problem was already solved after correcting TLS settings.

**Comment by gep13** [comment]
> @mcsmit said...
I would love to test but my problem was already solved after correcting TLS settings.

Apologies, I did read the comments that you made in this issue, I simply tagged everyone that had commented.  Happy to hear that your issue was already resolved.

**Comment by yuichi-github** [comment]
I've checked that the problem has been fixed in the alpha version (2.1.0-alpha-20230614).
I appreciate it very much !

**Comment by gep13** [comment]
> @yuichi-github said...
I've checked that the problem has been fixed in the alpha version (2.1.0-alpha-20230614).

Thank you very much for confirming!

**Comment by gep13** [comment]
Assuming you are talking about updating to the alpha release, can you try the command again, including the `--pre` option.

**Comment by gep13** [comment]
No, once installed, the alpha version number should be reported when running `choco --version` like the following:

<img width="284" alt="image" src="https://github.com/chocolatey/choco/assets/1271146/c6c9b41c-bd4a-401f-b17a-c57fb98acfb0">

Can you provide the exact command that you are running, as well as a screenshot, or listing, of the files that are stored in the folder that you placed the Chocolatey nupkg file.

**Comment by gep13** [comment]
Would you be able to provide the log files for the upgrade attempt?  Off the top of my head, I am not clear on what could be causing the issue that you are describing. Thanks

**Comment by gep13** [comment]
@opusforlife2 I am not seeing anything obvious in the log to indicate that there is a problem following the upgrade, so I am not sure why you are not seeing the choco.exe being updated to the alpha version.

As a quick check, in a PowerShell session, can you run the following:

```
&where.exe choco
```

And let me know what the output is?  Thanks

**Comment by gep13** [comment]
Thank you for confirming.  I don't have anything else to suggest at the minute.  Let me bring this up in a meeting that I have later today, and someone will reply back with any other suggestions that we have.

**Comment by ChristianKoessel** [comment]
I can confirm that my issue is solved with v2.1.0
Our corporate system proxy settings are automatically detected again.

**Comment by brandilton** [comment]
I'm getting the the same error on a device with no proxy at all and have fully removed chocolatey by deleting the $env:SYSTEMDRIVE\ProgramData\Chocolatey folder, then reinstalling by running Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1')).  The issue persists after a full reinstall to 2.1.0.  This is happening on multiple devices but not all.

choco outdated                                          
Chocolatey v2.1.0                                                               
Outdated Packages                                                               
 Output is package name | current version | available version | pinned?         
                                                                                
Unable to load the service index for source https://community.chocolatey.org/api
/v2/.                                                                           
Unable to connect to source 'https://community.chocolatey.org/api/v2/':         
 Object reference not set to an instance of an object.                          
chocolatey|2.1.0|2.1.0|false                                                    
                                                                                
Chocolatey has determined 0 package(s) are outdated.                            
 1 package(s) had warnings.                                                     
Warnings:                                                                       
 - chocolatey

**Comment by corbob** [comment]
@brandilton it would be best to open a new issue and complete the template so we can get further details like the logs to assist you.
